client/app/visualizations/chart/plotly/utils.js

```diff
@@ -380,6 +378,18 @@ export function prepareLayout(element, seriesList, options, data) {
     }
   }
 
+  // Truncate x-axis labels using ticktext.
+  let ticktext, tickvals;
+  const xAxisLabelLength = parseInt(options.xAxisLabelLength, 10) || DEFAULT_XAXIS_LABEL_LENGTH;
+  if (data.length > 0) {
+    ticktext = map(data[0].x, xVal => String(xVal).substr(0, xAxisLabelLength));
+    tickvals = data[0].x;
+  }
+  if (ticktext && tickvals) {
+    result.xaxis.ticktext = ticktext;
+    result.xaxis.tickvals = tickvals;
```

Comment:
Looks good to me, I didn't know about `ticktext`, looking at https://plot.ly/javascript/axes/#enumerated-ticks-with-tickvals-and-ticktext made it more clear what it looks like. Maybe add a comment on where that comes from?